packages/flutter/lib/src/material/page.dart

```diff
@@ -142,17 +145,27 @@ class _CupertinoBackGestureController extends NavigationGestureController {
 
     // Don't end the gesture until the transition completes.
     final AnimationStatus status = controller.status;
-    handleStatusChanged(controller.status);
+    handleStatusChanged(status);
     controller?.addStatusListener(handleStatusChanged);
 
     return (status == AnimationStatus.reverse || status == AnimationStatus.dismissed);
   }
 
   void handleStatusChanged(AnimationStatus status) {
-    if (status == AnimationStatus.dismissed)
+    // This can happen if an earlier status listener ends up calling dispose()
+    // on this object.
+    // TODO(abarth): Consider changing AnimationController not to call listeners
+    // that were removed while calling other listeners.
+    if (controller == null)
+      return;
+
+    if (status == AnimationStatus.dismissed) {
       navigator.pop();
-    if (status == AnimationStatus.dismissed || status == AnimationStatus.completed)
+      assert(controller == null);
+    } else if (status == AnimationStatus.completed) {
       dispose();
```

Comment:
Yes, I had a version of this patch that went further and tried to clean up this navigation gesture stuff, but I decided to try to fix one problem at a time.